packages/trees/frontend/sirius-components-trees/src/trees/useTreeSelection.tsx

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import { useCallback, useState } from 'react';
+import { GQLTree, GQLTreeItem } from '../views/TreeView.types';
+import { TreeItemClickResult, UseTreeSelectionValue } from './useTreeSelection.types';
+
+/**
+ * Hook to handle the different scenarios of tree item selection: simple click, multi-selection (CTRL), range selection (SHIFT).
+ * @returns An object containing the function to handle tree item click events.
+ */
+export const useTreeSelection = (): UseTreeSelectionValue => {
+  const [pivotId, setPivotId] = useState<string | null>(null);
+
+  const computeSelectedElementsInRange = useCallback(
```

Comment:
- Why useCallback?
- Why is that function defined twice, once here and once in the explorer view?